packages/veritone-react-common/src/components/SourceManagementForm/index.js

```diff
@@ -0,0 +1,267 @@
+      const sourceTypeIdx = 0;
+      const properties = get(sourceTypes, [
+        sourceTypeIdx,
+        'sourceSchema',
+        'definition',
+        'properties'
+      ]);
+
+      if (properties) {
+        Object.keys(properties).forEach(field => {
+          fieldValues[field] = '';
+        });
+      }
+
+      newState.sourceConfig = {
+        ...this.state.sourceConfig,
+        sourceTypeId: sourceTypes[sourceTypeIdx].id,
+        details: {
+          ...fieldValues
+        }
+      };
+    }
+
+    return this.setState(newState);
+  }
+
+  handleOnClose = () => {
+    return this.setState({ openDialog: false }, () => {
+      this.props.onClose();
+    });
+  };
+
+  handleChangeTab = (e, tabIdx) => {
+    return this.setState({ activeTab: tabIdx });
+  };
+
+  saveConfiguration = config => {
+    return this.setState(prevState => ({
+      sourceConfig: {
+        ...prevState.sourceConfig,
+        ...config
+      }
+    }));
+  };
+
+  manageTemplatesList = (templateSchemaId, remove = false) => {
+    const { templateData, initialTemplates } = this.props;
+
+    if (remove) {
+      if (this.state.contentTemplates[templateSchemaId]) {
+        return this.setState(prevState => {
+          const contentTemplates = { ...prevState.contentTemplates };
+          delete contentTemplates[templateSchemaId];
+
+          return { contentTemplates };
+        });
+      }
+    } else {
+      const data = {};
+      Object.keys(templateData[templateSchemaId].definition.properties).reduce(
+        (fields, schemaDefProp) => {
+          data[schemaDefProp] =
+            initialTemplates[templateSchemaId] &&
+              initialTemplates[templateSchemaId].data
+              ? initialTemplates[templateSchemaId].data[schemaDefProp]
+              : '';
+        },
+        data
+      );
+
+      this.setState(prevState => ({
+        contentTemplates: {
+          [templateSchemaId]: {
+            ...templateData[templateSchemaId],
+            data
+          },
+          ...prevState.contentTemplates
+        }
+      }));
+    }
+  };
+
+  updateTemplateDetails = (templateSchemaId, fieldId, value) => {
+    const { contentTemplates } = this.state;
+
+    this.setState({
+      contentTemplates: {
+        ...contentTemplates,
+        [templateSchemaId]: {
+          ...contentTemplates[templateSchemaId],
+          data: {
+            ...contentTemplates[templateSchemaId].data,
+            [fieldId]: value
+          }
+        }
+      }
+    });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    return this.props.onSubmit({
+      sourceConfiguration: this.state.sourceConfig,
+      contentTemplates: this.state.contentTemplates
+    });
+  };
+
+  render() {
+    const { activeTab } = this.state;
+
+    return (
+      <FullScreenDialog open={this.state.openDialog}>
+        <div className={styles['sm-form-wrapper']}>
+          <ModalHeader
+            title={
+              this.state.sourceConfig.name
```

Comment:
this.state.sourceConfig.name || 'New Source'

Reply:
fixed